data/projects.js

```diff
@@ -147,5 +147,119 @@ coala should support generating metrics for your code.",
 		"initiatives" : ["GSoC"],
 		"tags" : ["Web", "Blog"],
 		"markdown": "coala_web.md"
+	},
+	{
+		"name" : "Optimize Caching",
+		"desc" : "Improve coala's performance by implementing caching strategies.",
+		"requirements" : [
+			"At least one patch to the coala core should be accepted and merged.",
+			"The student should be familiar with ongoing technical cEPs, \
+			especially <https://github.com/coala/cEPs/blob/master/cEP-0002.md>",
+			"The proposal needs to show some familiarity with coala's caching \
+			implementation.",
+		],
+		"difficulty" : "medium",
+		"issues" : [
+			"https://github.com/coala/coala/issues/2541",
+			"https://github.com/coala/coala/issues/2699",
+			"https://github.com/coala/coala/issues/2912"
+		],
+		"mentors" : [
+			"adtac",
+			"sils"
+		],
+		"initiatives" : ["GSoC"],
+		"tags" : ["Core", "Bears"],
+		"markdown": "optimize_caching.md"
+	},
+	{
+		"name" : "Cohesive Editor Support",
+		"desc" : "Develop an efficient strategy that allows coala to maintain \
+		a number of plugins for different editors.",
+		"requirements" : [
+			"The proposal has to provide a convincing strategy for testin, \
+			 documentation and maintaining editor plugins",
+			"The applicant should have one patch accepted to any of the coala \
+			plugins."
+		],
+		"difficulty" : "medium",
+		"issues" : [
+			"https://github.com/coala/coala-atom/issues/37",
+			"https://github.com/coala/coala-emacs/issues/11",
+			"https://github.com/coala/coala-eclipse/issues/25",
+			"https://github.com/coala/coala-vim/issues/24",
+			"https://github.com/coala/coala-gedit/issues/5",
+			"https://github.com/coala/coala-sublime/issues/4"
+		],
+		"mentors" : [
+			"AbdealiJK",
+			"fneu"
+		],
+		"initiatives" : ["GSoC"],
+		"tags" : ["Editors", "Plugins"],
+		"markdown": "editor_support.md"
+	},
+	{
+		"name" : "Python dead code fixer",
+		"desc" : "Wrap two flake8 AST plugins and write a native bear for dead \
+		code detection and fixing for python.",
+		"requirements" : [
+			"At least one patch to the coala core should be accepted and merged.",
+		],
+		"difficulty" : "medium",
+		"issues" : [],
+		"mentors" : [
+			"jayvdb"
+		],
+		"initiatives" : ["GSoC"],
+		"tags" : ["Bears"],
+		"markdown": "dead_python_code.md"
+	},
+	{
+		"name" : "Real-time team communication bridge",
+		"desc" : "A way to post CI problems to online team rooms, such as IRC, \
+		Telegram, slack or gitter.",
+		"requirements" : [
+			"At least one patch to the coala core should be accepted and merged.",
```

Comment:
I have no objection to a cobot patch as the requirement; that is also a good way for candidates to prepare and for us to evaluate.

But this isnt a cobot project. It is requiring hooking up coala core on a VM with a 'room' of some sort, where the room is dedicated to cleanup of a repo which doesnt yet have CI enforcing quality. Imagine a huge repo with thousands of coala errors, and bot in the room tries to keep the cleanup ball rolling, advertising new problems as they arise, and trying to allocate old problems to contributors.  E.g. a 'newcomer' bug generator, explaining the problem and how to fix it.